src/components/Button/Button.tsx

```diff
@@ -292,12 +315,23 @@ export const Button: FC<ButtonProps> = React.forwardRef(
     ): JSX.Element => (
       <span className={buttonTextClassNames}>
         {text ? text : 'Button'}
-        {counterExists && <Badge classNames={badgeClassNames}>{counter}</Badge>}
+        {counterExists && (
+          <Badge
+            {...badgeProps}
+            classNames={mergeClasses([badgeClassNames, badgeProps?.classNames])}
+          >
+            {counter}
+          </Badge>
+        )}
       </span>
     );
 
     return (
-      <>
+      <ThemeContextProvider
```

Comment:
For the most part, yes.

Reply:
This is how we create the pointer.

Reply:
The way we avoid this cannot be done until Octuple v3.0.0 - where we might choose to use the `Atom` component for every  base. This would be one way to consolidate all contextual properties: disabled, gradient, theme, and size. However, this will be a much more risky change than a simple wrapper.